### PR TITLE
More updates to zh_CN translations

### PR DESCRIPTION
**Pull request by vadz**
@hjiawei Could you please confirm that there are no problems with the latest translations changes from @obmij ?

### CODE REVIEW

File: locale/zh_CN.po [hunk 1]
```diff
@@ -1027,12 +1027,12 @@ msgstr "无法创建调试报告。"
 #: ../src/common/debugrpt.cpp:228
 #, c-format
 msgid "Failed to remove debug report file \"%s\""
-msgstr "无法删除调试报告文件 \"%s\""
+msgstr "无法删除调试报告文件 \"%s\"。"
```
Comment (hjiawei) [inline locale/zh_CN.po:1030]: nit. no period `.` in the original English text. I would like to remove the `。` to be consistent. Nothing is wrong though. That is probably just my personal preference.

File: locale/zh_CN.po [hunk 2]
```diff
@@ -1027,12 +1027,12 @@ msgstr "无法创建调试报告。"
...
 
 #: ../src/common/debugrpt.cpp:240
 #, c-format
 msgid "Failed to clean up debug report directory \"%s\""
-msgstr "无法清除调试报告目录 \"%s\""
+msgstr "无法清除调试报告目录 \"%s\"。"
```
Comment (hjiawei) [inline locale/zh_CN.po:1035]: nit. no period `.` in the original English text. I would like to remove the `。` to be consistent. Nothing is wrong though. That is probably just my personal preference.

File: locale/zh_CN.po [hunk 3]
```diff
@@ -1050,11 +1050,11 @@ msgstr "无法生成调试报告。"
 #, c-format
 msgid ""
 "Processing debug report has failed, leaving the files in \"%s\" directory."
-msgstr "处理调试报告失败, 文件被保存在目录 \"%s\" 中。"
+msgstr "处理调试报告失败, 文件被保存在 \"%s\" 目录。"
 
 #: ../src/common/debugrpt.cpp:574
 msgid "A debug report has been generated. It can be found in"
-msgstr "产生了一份调试报告。 位于"
+msgstr "产生了一份调试报告，位于"
```
Comment (hjiawei) [inline locale/zh_CN.po:1057]: Same here. The original text is a period `.` rather than a comma `,`.

File: locale/zh_CN.po [hunk 4]
```diff
@@ -1200,12 +1200,12 @@ msgstr "无法关闭文件 '%s'"
 #: ../src/common/ffile.cpp:107 ../src/common/ffile.cpp:132
 #, c-format
 msgid "Read error on file '%s'"
-msgstr "读文件 '%s' 出错"
+msgstr "读文件 '%s'出错"
```
Comment (hjiawei) [inline locale/zh_CN.po:1203]: nit. spacing.

File: locale/zh_CN.po [hunk 5]
```diff
@@ -1358,7 +1358,7 @@ msgstr "无法更新用户配置文件。"
 
 #: ../src/common/fileconf.cpp:1029
 msgid "Error saving user configuration data."
-msgstr "保存用户配置数据错误。"
+msgstr "保存用户配置数据错误."
```
Comment (hjiawei) [inline locale/zh_CN.po:1361]: revert this. I don't think `.` is correct.

File: locale/zh_CN.po [hunk 6]
```diff
@@ -1378,22 +1378,22 @@ msgstr "忽略对只读键 '%s' 的修改。"
 #: ../src/common/fileconf.cpp:1946
 #, c-format
 msgid "trailing backslash ignored in '%s'"
-msgstr "'%s' 尾部的斜线将被忽略"
+msgstr "'%s'尾部的斜线将被忽略"
```
Comment (hjiawei) [inline locale/zh_CN.po:1381]: nit. space.

File: locale/zh_CN.po [hunk 7]
```diff
@@ -2128,7 +2129,7 @@ msgstr "BMP: 无法写数据。"
 #: ../src/common/imagbmp.cpp:533
 #, c-format
 msgid "BMP: header has biClrUsed=%d when biBitCount=%d."
-msgstr "BMP: 当 biBitCount=%d 时，文件头为 biClrUsed=%d。"
+msgstr "BMP: 当 biBitCount=%d 时，表头則為 biClrUsed=%d。"
```
[the remark this answers is not in the document]
Comment (obmij) [reply locale/zh_CN.po:2132]: Screen Shot 2022-07-06 at 12 49 24 AM

File: locale/zh_CN.po (hunk 3 above)
Comment (obmij) [reply locale/zh_CN.po:1057]: replaced with ，so the context connects. This is human translation, not MT.

File: locale/zh_CN.po [hunk 8]
```diff
@@ -1070,7 +1070,7 @@ msgstr ""
 
 #: ../src/common/debugrpt.cpp:730
 msgid "Failed to execute curl, please install it in PATH."
-msgstr "无法执行 curl，请在 PATH 变量所指的目录中安装它。"
+msgstr "无法执行curl，请在 PATH 变量所指的目录中安装curl。"
```
[the remark this answers is not in the document]
Comment (obmij) [reply locale/zh_CN.po:1073]: yes, a space is required between two different languages.

**Comment by obmij** [comment]
There are many Halfwidth and fullwidth forms of punctuation mixed together which is very unlikely seen.
e.g. （）()

**Comment by obmij** [comment]
space, punctuation, symbol inconsistency - (partial)
msgstr "前页(&P)"
msgstr "无法撤销(&U) "
msgstr "阴影颜色 (&o)："
msgstr "阴影扩散 (&a):"
msgstr "模糊距离 (&B):"

File: locale/zh_CN.po [hunk 9]
```diff
@@ -4528,15 +4529,15 @@ msgstr "允许"
 
 #: ../src/generic/filectrlg.cpp:429
 msgid "Attributes"
-msgstr "属性"
+msgstr "属性(Attributes)"
```
Comment (hjiawei) [inline locale/zh_CN.po:4532]: Why do we need the English word in parentheses when it is translated?

File: locale/zh_CN.po [hunk 10]
```diff
@@ -4634,23 +4635,23 @@ msgstr "字体样式。"
...
+msgstr "字体粗细(&W):"
 
 #: ../src/generic/fontdlgg.cpp:392 ../src/generic/fontdlgg.cpp:394
 msgid "The font weight."
 msgstr "字体粗细。"
 
 #: ../src/generic/fontdlgg.cpp:402
 msgid "C&olour:"
-msgstr "颜色 (&o):"
+msgstr "颜色(&o):"
```
Comment (hjiawei) [inline locale/zh_CN.po:4646]: nit. Capitalise `o`?

File: locale/zh_CN.po [hunk 11]
```diff
@@ -5247,7 +5248,7 @@ msgstr "显示全部"
 msgid ""
 "Display all index items that contain given substring. Search is case "
 "insensitive."
-msgstr "显示包含给定子串的所有索引项. 搜索是大小写无关的。"
+msgstr "显示包含给定子串的所有索引项. 搜索是大小写无关的."
```
Comment (hjiawei) [inline locale/zh_CN.po:5251]: nit. `。`?

File: locale/zh_CN.po [hunk 12]
```diff
@@ -5356,23 +5357,23 @@ msgstr "字体大小"
...
 
 #: ../src/html/helpwnd.cpp:1263
 msgid "<b><i>Bold italic face.</i></b><br>"
-msgstr "<b><i>粗斜体。</i></b><br>"
+msgstr "<b><i>粗斜体.</i></b><br>"
 
 #: ../src/html/helpwnd.cpp:1266
 msgid "Fixed size face.<br> <b>bold</b> <i>italic</i> "
-msgstr "固定字体。<br> <b>粗体</b> <i>斜体</i> "
+msgstr "固定字体. <br> <b>粗体</b> <i>斜体</i> "
```
Comment (hjiawei) [inline locale/zh_CN.po:5376]: nit. nit. `。`?

File: locale/zh_CN.po [hunk 13]
```diff
@@ -5410,7 +5411,7 @@ msgstr "%i / %u"
 #: ../src/html/helpwnd.cpp:1713
 #, c-format
 msgid "%u of %u"
-msgstr "%u / %u"
+msgstr "%u 的 %u"
```
Comment (hjiawei) [inline locale/zh_CN.po:5414]: This is different from the one above. Can we make these two consistent (either `/` or `的` is good to me)?

File: locale/zh_CN.po [hunk 14]
```diff
@@ -5442,7 +5443,7 @@ msgstr "HTML 锚 %s 不存在。"
 #: ../src/html/htmlwin.cpp:1020
 #, c-format
 msgid "Copied to clipboard:\"%s\""
-msgstr "已复制到剪贴板: \"%s\""
+msgstr "已复制到剪贴板:\"%s\""
```
Comment (hjiawei) [inline locale/zh_CN.po:5446]: nit. space?

File: locale/zh_CN.po [hunk 15]
```diff
@@ -5840,7 +5841,7 @@ msgstr "当前不支持监视单独文件的更改。"
 #: ../src/msw/fswatcher.cpp:165
 #, c-format
 msgid "Unable to set up watch for '%s'"
-msgstr "无法为 '%s' 设置监视"
+msgstr "无法为 '%s' 设置 watch"
```
Comment (hjiawei) [inline locale/zh_CN.po:5844]: I see some other places are using `监视` instead of untranslated `watch`. WDYT?

File: locale/zh_CN.po [hunk 16]
```diff
@@ -6353,32 +6354,32 @@ msgstr "<任意>"
...
 #: ../src/osx/carbon/fontdlg.cpp:652
 msgid "<Any Swiss>"
-msgstr "<任意瑞士>"
+msgstr "<任意 Swiss>"
 
 #: ../src/osx/carbon/fontdlg.cpp:422 ../src/osx/carbon/fontdlg.cpp:630
 #: ../src/osx/carbon/fontdlg.cpp:650
 msgid "<Any Teletype>"
-msgstr "<任意打字>"
+msgstr "<任意 Teletype>"
```
Comment (hjiawei) [inline locale/zh_CN.po:6382]: If you search any of these (like `Modern` or Teletype), there are places where these are translated. Probably we can also make them consistent?

File: locale/zh_CN.po [hunk 17]
```diff
@@ -6443,7 +6444,7 @@ msgstr "隐藏 %s"
 #: ../src/osx/menu_osx.cpp:517
 msgctxt "macOS menu item"
 msgid "Hide Application"
-msgstr "隐藏应用程序"
+msgstr "应用程序"
```
Comment (hjiawei) [inline locale/zh_CN.po:6447]: Hide is not translated (or reverted).

File: locale/zh_CN.po [hunk 18]
```diff
@@ -6464,7 +6465,7 @@ msgstr "退出 %s"
 #: ../src/osx/menu_osx.cpp:531
 msgctxt "macOS menu item"
 msgid "Quit Application"
-msgstr "退出应用程序"
+msgstr "应用程序"
```
Comment (hjiawei) [inline locale/zh_CN.po:6468]: `Quit` is not translated (or removed).

File: locale/zh_CN.po [hunk 19]
```diff
@@ -6497,7 +6498,7 @@ msgstr "字体粗细"
 #. TRANSLATORS: Label of font family
 #: ../src/propgrid/advprops.cpp:605
 msgid "Family"
-msgstr "系列"
+msgstr "系列(&F)"
```
Comment (hjiawei) [inline locale/zh_CN.po:6501]: No keyboard shortcut in the original English message.

File: locale/zh_CN.po [hunk 20]
```diff
@@ -6602,17 +6603,17 @@ msgstr "工具提示文字"
...
 #: ../src/propgrid/advprops.cpp:796
 msgid "WindowFrame"
-msgstr "窗框"
+msgstr "窗框(&W)"
 
 #. TRANSLATORS: Keyword of system colour
 #: ../src/propgrid/advprops.cpp:797
 msgid "WindowText"
-msgstr "视窗文字"
+msgstr "视窗文字(&W)"
```
Comment (hjiawei) [inline locale/zh_CN.po:6616]: No keyboard shortcut in the original English message.

File: locale/zh_CN.po (hunk 11 above)
Comment (obmij) [reply locale/zh_CN.po:5251]: There we go again... my delivery on June 25th, 2022 used 。
Screen Shot 2022-07-06 at 2 02 54 AM

File: locale/zh_CN.po [hunk 21]
```diff
@@ -6690,7 +6691,7 @@ msgstr "黄色"
 
 #: ../src/propgrid/advprops.cpp:1518
 msgid "White"
-msgstr "白色"
+msgstr "百色"
```
Comment (hjiawei) [inline locale/zh_CN.po:6694]: This is not correct.

File: locale/zh_CN.po [hunk 22]
```diff
@@ -6770,12 +6771,12 @@ msgstr "铅笔"
...
 msgctxt "system cursor name"
 msgid "Point Left"
-msgstr "向左指"
+msgstr "向左对齐"
 
 #: ../src/propgrid/advprops.cpp:1670
 msgctxt "system cursor name"
 msgid "Point Right"
-msgstr "向右指"
+msgstr "向右对齐"
```
Comment (hjiawei) [inline locale/zh_CN.po:6779]: These are more like `指` (`Point`) rather than `对齐` (Align) IMO.

File: locale/zh_CN.po [hunk 23]
```diff
@@ -6790,32 +6791,32 @@ msgstr "右键"
...
 msgctxt "system cursor name"
 msgid "Sizing NW-SE"
-msgstr "缩放 (东南至西北)"
+msgstr "尺寸 NW-SE"
 
 #: ../src/propgrid/advprops.cpp:1676
 msgctxt "system cursor name"
 msgid "Sizing W-E"
-msgstr "缩放 (东至西)"
+msgstr "尺寸 W-S"
```
Comment (hjiawei) [inline locale/zh_CN.po:6809]: These are actually taken from the `zh-TW.po`. Any reasons to revert these?

File: locale/zh_CN.po (hunk 16 above)
Comment (obmij) [reply locale/zh_CN.po:6382]: oh definitely... how about "任一XX"

File: locale/zh_CN.po [hunk 24]
```diff
@@ -6946,7 +6947,7 @@ msgstr "阴影"
 
 #: ../src/richtext/richtextbackgroundpage.cpp:194
 msgid "Use &shadow"
-msgstr "使用阴影 (&s)"
+msgstr "用(&S)"
```
Comment (hjiawei) [inline locale/zh_CN.po:6950]: `shadow` is not translated.

File: locale/zh_CN.po [hunk 25]
```diff
@@ -7161,7 +7162,7 @@ msgstr "阴影扩散。"
 
 #: ../src/richtext/richtextbackgroundpage.cpp:298
 msgid "&Blur distance:"
-msgstr "模糊距离 (&B):"
+msgstr "模糊距离(&A)："
```
Comment (hjiawei) [inline locale/zh_CN.po:7165]: This shortcut is `B`.

File: locale/zh_CN.po [hunk 26]
```diff
@@ -7329,7 +7330,7 @@ msgstr "角"
 
 #: ../src/richtext/richtextborderspage.cpp:575
 msgid "Corner &radius:"
-msgstr "角半径 (&r):"
+msgstr "角&半径:"
```
Comment (hjiawei) [inline locale/zh_CN.po:7333]: Missing keyboard shortcuts `(&R)`.

File: locale/zh_CN.po [hunk 27]
```diff
@@ -7427,7 +7428,7 @@ msgstr "插入范围"
 
 #: ../src/richtext/richtextbuffer.cpp:8410
 msgid "Too many EndStyle calls!"
-msgstr "调用 EndStyle 太多次！"
+msgstr "呼叫 EndStyle 太多次！"
```
Comment (hjiawei) [inline locale/zh_CN.po:7431]: I think `调用` is more accurate that `呼叫` in this case.

File: locale/zh_CN.po (hunk 23 above)
Comment (obmij) [reply locale/zh_CN.po:6809]: I had no idea... zh_TW.po looks like this
Screen Shot 2022-07-06 at 2 16 18 AM

File: locale/zh_CN.po [hunk 28]
```diff
@@ -8695,20 +8696,20 @@ msgstr "符号使用该字体。"
...
 #: ../src/richtext/richtextsymboldlg.cpp:410
 msgid "&Subset:"
-msgstr "子集 (&S):"
+msgstr "子集(&S):"
 
 #: ../src/richtext/richtextsymboldlg.cpp:418
 #: ../src/richtext/richtextsymboldlg.cpp:420
 msgid "Shows a Unicode subset."
-msgstr "显示 Unicode 子集。"
+msgstr "显示Unicode子集。"
```
Comment (hjiawei) [inline locale/zh_CN.po:8704]: nit. space in between.

File: locale/zh_CN.po (hunk 20 above)
Comment (obmij) [reply locale/zh_CN.po:6616]: I suppose a full check is required, the later the better.
Screen Shot 2022-07-06 at 2 18 24 AM

**Comment by hjiawei** [comment]
Thanks for the great work @obmij. Overall LGTM. There are some minor issues that I commented inline. Most others are nitpicking. You can either take it or leave it as you see fit (my comments are not 100% correct either). Thanks again.

File: locale/zh_CN.po (hunk 10 above)
Comment (obmij) [reply locale/zh_CN.po:4646]: Here's the translation style example -
CB(btn_exit,Exit,Exit()) --> CB(btn_exit,結束\(&E),Exit())

File: locale/zh_CN.po (hunk 23 above)
Comment (obmij) [reply locale/zh_CN.po:6809]: > Thanks for the great work @obmij. Overall LGTM. There are some minor issues that I commented inline. Most others are nitpicking. You can either take it or leave it as you see fit (my comments are not 100% correct either). Thanks again.



> Thanks for the great work @obmij. Overall LGTM. There are some minor issues that I commented inline. Most others are nitpicking. You can either take it or leave it as you see fit (my comments are not 100% correct either). Thanks again.

@hjiawei Pleasure collaboration! How about giving me some time to dig out these symbols and pieces of stuff. I assume it's no guilt for deferring a few days? ;-)

**Comment by hjiawei** [comment]
BTW, I suspect there are some git merge conflicts between PR https://github.com/wxWidgets/wxWidgets/pull/22571 and your changes in https://github.com/wxWidgets/wxWidgets/commit/d4f5b00a8c56fa4c31f0152f553d1ac3f5e50de5 so some of your fixes got bad merges. That might explain your fixes got reverted or removed or whatsoever.

**Comment by vadz** [comment]
Sorry, I didn't really follow ~50 comments on this one, so could you please just let me know if this should be merged in its current state or, if not, when should it be good to merge it? TIA!

**Comment by obmij** [comment]
I don’t think this one is good enough to release. Do allow me to re-compare and check again from raw texts, I’ll return to you as soon as I can.

**Comment by vadz** [comment]
Thanks, I'm keeping this one open in case you still need it, but if you don't, please let me know and let's simply close it and you will open another PR when you're ready.

**Comment by vadz** [comment]
Replaced by #22621.

**Pull request closed without merging**